applications/OptimizationApplication/python_scripts/controls/control.py

```diff
@@ -0,0 +1,114 @@
+from abc import ABC, abstractmethod
+from KratosMultiphysics.OptimizationApplication.utilities.union_utilities import SupportedSensitivityFieldVariableTypes
+from KratosMultiphysics.OptimizationApplication.utilities.union_utilities import ContainerExpressionTypes
+
+class Control(ABC):
+    """Base abstract control class.
+
+    The base abstract control class has the following responsibilities.
+        1. Initalizes and finalizes the control.
+        2. Mapping physical space gradients to control space gradients.
+        2. Updating the controlled parts of the model part with new design given in control space.
+        3. Retrieve control field.
+        4. Retrieve physical space kratos variables (If more than one physical space kratos variables are controlled by the given control field.)
+
+    This control should only work on one model part and one kratos control variable. Hence, if multiple model parts required then,
+    a single model part should be created using Kratos.ModelPartOperationUtilities.
+
+    """
+    @abstractmethod
+    def Initialize(self) -> None:
+        """Initializes the control.
+
+        This method initializes the control. This is only called once in the whole optimization process.
+
+        """
+        pass
+
+    @abstractmethod
+    def Check(self) -> None:
```

Comment:
I guess, one should implement it in individual control implementations if you need to check smth.

Reply:
Only use case for this is in the following situation, when a control has to do check once all the components (responses, execiton_policies, other controls) are properly initialized. Otherwise, we can do the check in Initialize it self. 

Whats your decision @RezaNajian @Igarizza ?